File: app/src/main/java/com/dev/bins/note/adapter/MainFragmentRecycleViewAdapter.java
package com.dev.bins.note.adapter;

import android.content.Context;
import android.content.Intent;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.dev.bins.note.R;
import com.dev.bins.note.model.Category;
import com.dev.bins.note.model.Note;
import com.dev.bins.note.ui.DetailActivity;

import org.litepal.crud.DataSupport;

import java.text.SimpleDateFormat;
import java.util.List;

/**
 * Created by bin on 11/24/15.
 */
public class MainFragmentRecycleViewAdapter extends RecyclerView.Adapter<Holder> {


    private SimpleDateFormat sdf;
    private Context context;
    List<Note> notes;


    public MainFragmentRecycleViewAdapter(Context context) {
        this.context = context;
        sdf = new SimpleDateFormat("yyyy年MM月dd日 HH:mm:ss");
        notes = DataSupport.where("isshow=? and category_id=?", "1",String.valueOf(Category.DEFAULT)).order("date desc").find(Note.class);
    }

    @Override
    public Holder onCreateViewHolder(ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.recycleview_item, parent, false);
        return new Holder(view);
    }

    @Override
    public void onBindViewHolder(Holder holder, final int position) {

        final Note note = notes.get(position);
        String title = note.getTitle();
        if (title.equals("无标题")) {
            holder.title.setText(String.valueOf(note.getId()));
        } else {
            holder.title.setText(title);
        }
        holder.content.setText(note.getContent());
        String date = sdf.format(note.getDate());
        holder.time.setText(date);
        holder.cardView.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {

                Intent intent = new Intent(context, DetailActivity.class);
                intent.putExtra("id", note.getId());
                context.startActivity(intent);
            }
        });
    }

    @Override
    public int getItemCount() {
        return notes.size();
    }

    public void add(List<Note> data) {
        notes.clear();
        notes.addAll(data);
        notifyDataSetChanged();
    }

//    class Holder extends RecyclerView.ViewHolder{
//        TextView title;
//        TextView content;
//        TextView time;
//        MaterialRippleLayout cardView;
//
//        public Holder(View itemView) {
//            super(itemView);
//            title = (TextView) itemView.findViewById(R.id.title);
//            content = (TextView) itemView.findViewById(R.id.content);
//            time = (TextView) itemView.findViewById(R.id.time);
//            cardView = (MaterialRippleLayout) itemView.findViewById(R.id.cardView);
//
//        }
//    }
}

File: app/src/main/java/com/dev/bins/note/adapter/MainFragmentRecycleViewAdapter.kt
package com.dev.bins.note.adapter

import android.content.Context
import android.content.Intent
import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.ViewGroup
import com.dev.bins.note.R
import com.dev.bins.note.model.Note
import com.dev.bins.note.ui.DetailActivity
import java.text.SimpleDateFormat
import java.util.*

/**
 * Created by bin on 11/24/15.
 */
class MainFragmentRecycleViewAdapter(private val context: Context) : RecyclerView.Adapter<Holder>() {


    private val sdf: SimpleDateFormat
    internal var notes: MutableList<Note>


    init {
        sdf = SimpleDateFormat("yyyy年MM月dd日 HH:mm:ss")
        notes = ArrayList()
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): Holder {
        val view = LayoutInflater.from(parent.context).inflate(R.layout.recycleview_item, parent, false)
        return Holder(view)
    }

    override fun onBindViewHolder(holder: Holder, position: Int) {

        val note = notes[position]
        val title = note.title
        if (title == "无标题") {
            holder.title.text = note.id.toString()
        } else {
            holder.title.text = title
        }
        holder.content.text = note.content
        val date = sdf.format(note.date)
        holder.time.text = date
        holder.cardView.setOnClickListener {
            val intent = Intent(context, DetailActivity::class.java)
            intent.putExtra("id", note.id)
            context.startActivity(intent)
        }
    }

    override fun getItemCount(): Int {
        return notes.size
    }

    fun add(data: List<Note>) {
        notes.clear()
        notes.addAll(data)
        notifyDataSetChanged()
    }

    //    class Holder extends RecyclerView.ViewHolder{
    //        TextView title;
    //        TextView content;
    //        TextView time;
    //        MaterialRippleLayout cardView;
    //
    //        public Holder(View itemView) {
    //            super(itemView);
    //            title = (TextView) itemView.findViewById(R.id.title);
    //            content = (TextView) itemView.findViewById(R.id.content);
    //            time = (TextView) itemView.findViewById(R.id.time);
    //            cardView = (MaterialRippleLayout) itemView.findViewById(R.id.cardView);
    //
    //        }
    //    }
}
